core/src/validator.rs

```diff
@@ -3030,6 +3033,51 @@ pub fn is_snapshot_config_valid(snapshot_config: &SnapshotConfig) -> bool {
     }
 }
 
+/// Based on the current feature flag activation and genesis certificate account in the root bank,
+/// determine which phase of the migration we are in and initialize accordingly.
+fn initialize_migration_status(my_pubkey: Pubkey, root_bank: Arc<Bank>) -> MigrationStatus {
```

Comment:
move to `migration.rs`?

Reply:
ah, yeah, makes sense. I'm not going to be too picky about it given this should get ripped out from the code in less than a year anyways